### PR TITLE
Completed Homepage using react router and used useEffect for api calls

### PR DESCRIPTION
**Pull request by dev-ashishsethi**
image

Files for review: 

1. Home.jsx
2. App.js

Please review😊

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ function App() {
       <Routes>
         <Route path="/" element={<Home/>}/>
         <Route path="/ProductListing" element={<ProductListing/>}/>
+
```
Comment (m26unkwn) [inline src/App.js:16]: make path name of productlisting to only products or all products

**Review by m26unkwn — APPROVED** [review]
File Structure is good till now.

File: src/App.js [hunk 2]
```diff
@@ -1,41 +1,22 @@
 import "./App.css";
-import logo from "./logo.png";
-
+import {Nav} from "./pages/Nav/Nav"
+import {Home} from "./pages/Home/Home.jsx"
+import {Route,Link,Routes} from "react-router-dom"
```
Comment (Debarshi95) [inline src/App.js:4]: Add whitespace between the brackets and commas. Proper formatting of your code is a good thing. Update your vscode settings and set some default formatter like Prettier.

File: src/pages/Home/home.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            } catch (error) {
+                console.error(error);
+            }
+        })();
+        
+        let i=1;
+            setInterval(() => {
+                setCarousel(carousel[i].image);
+                i+=1;
+                i>3?i=1:i=i;
```
Comment (Debarshi95) [inline src/pages/Home/home.jsx:27]: Is the logic efficient ```i=i```?

File: src/pages/Home/home.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            try {
+                const response=await axios.get("/api/categories");
+                setData(response.data.categories);
+            } catch (error) {
+                console.error(error);
+            }
+        })();
+        
+        let i=1;
+            setInterval(() => {
```
Comment (Debarshi95) [inline src/pages/Home/home.jsx:24]: Be careful, when you use **setInterval**. Always and always clear the interval using ```clearInterval``` when you unmount your component. 
  
Read more here: https://upmostly.com/tutorials/setinterval-in-react-components-using-hooks

**Review by Debarshi95 — COMMENTED** [review]
All good but do check the comments.

File: src/pages/Home/home.jsx (hunk 4 above)
Comment (dev-ashishsethi) [reply src/pages/Home/home.jsx:24]: Thanks Debarshi😊

**Pull request merged**